alibi/explainers/shap_wrappers.py

```diff
@@ -0,0 +1,1431 @@
+            #  return the shap values?
+            raise NotImplementedError(
+                "Interactions can currently only be computed if no background dataset is specified. "
+                "Re-instantiate the explainer and run fit without any arguments to compute shap "
+                "interaction values!")
+        if y is not None:
+            raise NotImplementedError(
+                "Interactions can currently only be computed if no background dataset is specified "
+                "but explaining loss functions requires a background dataset. Re-instantiate the "
+                "explainer with model_output='log_loss' and run fit(background_data=my_data) to "
+                "explain loss values!!"
+            )
+
+    def _check_explainer_setup(self,
+                               background_data: Union[np.ndarray, pd.DataFrame, None],
+                               model_output: str,
+                               y: Optional[np.ndarray]) -> None:
+        """
+        Checks whether the inputs to the `explain` method match the explainer setup if shap interaction values are
+        note required
+
+        Parameters
+        ----------
+        background_data
+            See `fit` method documentation.
+        model_output
+            See TreeShap constructor documentation.
+        y
+            See`explain` method documentation.
+
+
+        Raises
+        ------
+        NotImplementedError
+            If the users passes labels to the `explain` method but does not specify a background dataset or
+            if the user does not pass a background dataset to `fit` and specifies a `model_output` other than
+            raw when initialising the explainer.
+        ValueError
+            If the user passes labels to the `explain` method but has not set `model_output='log_loss'` when
+            initialising the explainer.
+        """
+        # check settings are correct for loss value explanations
+        if y is not None:
+            if background_data is None:
+                raise NotImplementedError(
+                    "Loss values can only currently be explained with respect to the value over a background dataset. "
+                    "Re-instantiate the explainer and run fit(background_data=my_data)!")
+            if model_output != 'log_loss':
+                raise ValueError(
+                    "Model output should be set to 'log_loss' in order to explain loss values. Re-instantiate the model"
+                    "with the option `model_output='log_loss' passed to the constructor, call  "
+                    "fit(background_data=my_data) and then explain with the default arguments."
+                )
+        # check model output data is compatible with background data setting
+        else:
+            if background_data is None:
+                # TODO: @ Janis: Technically, we can catch this error in fit, set model_output='raw' and warn
```

Comment:
Again, I'd rather raise an error than provide a warning and change the behaviour.